lib/scryfall/cards.rb

```diff
@@ -17,5 +17,10 @@ def self.named(query:, exact: false, set: nil)
       params[:set] = set unless set.nil?
       api.get(params)
     end
+
+    def self.random(query: ' ')
+      params = { path: '/cards/random', q: query.encode }
+      api.get(params)
+    end
```

Comment:
Nicely done!

Reply:
Thanks, i wasn't sure that (query: ' ') would be correct, but it works.

Reply:
Actually it might be better to do it like above where we have the default arg as nil and then set it only if it's not nil for the sake of consistency across methods.

Reply:
Yeah, i actually tried it that way, and it doesn't work for some reason.

Reply:
Never mind I just tried it again and it does work. I'm not sure what was different last time but it kept telling me that nil wasn't a good value, and the second I replaced it with the space it worked. I'll correct it now.

Reply:
Sorry, I forgot to save after changing it to nil. After I saved it now gives me an error. 

NoMethodError:
       undefined method `encode' for nil:NilClass

This comes up when I run the test, and trying to run in irb also gets me an error as well.

uninitialized constant Scryfall (NameError)

Reply:
params = { path: '/cards/random', q: query.encode }

I tried doing some searching on this, and I think the issue is that we can't set query to nil because we're calling .encode.

Reply:
Yeah, you need to do it like in the `named` method above:

`params["q"] = query.encode unless query.nil?`